packages/swap/src/quoting/Quoter.ts

```diff
@@ -58,28 +60,44 @@ export type RpcLimitOrder = {
   };
 };
 
-const formatLimitOrders = (
-  quotes: { orders: MarketMakerQuote['legs'][number] }[],
-  leg?: MarketMakerQuoteRequest<LegJson>['legs'][number],
-): RpcLimitOrder[] => {
-  if (!leg) return [];
-
-  return quotes.flatMap(({ orders }) =>
-    orders.map(([tick, amount]) => ({
-      LimitOrder: {
-        side: leg.side === 'BUY' ? 'sell' : 'buy',
-        base_asset: leg.base_asset,
-        quote_asset: leg.quote_asset,
-        tick,
-        sell_amount: hexEncodeNumber(amount),
-      },
-    })),
-  );
+const isBalanceWithinTolerance = (balance: bigint, amount: bigint) => {
+  const tolerance = BigInt(env.QUOTER_BALANCE_TOLERANCE_PERCENT);
+  let paddedBalance = balance;
+  if (tolerance > 0n) paddedBalance += (balance * tolerance) / 100n;
+  return paddedBalance >= amount;
 };
 
+function* formatLimitOrders(
+  quotes: { orders: MarketMakerQuote['legs'][number]; balances?: InternalAssetMap<bigint> }[],
+  leg?: MarketMakerQuoteRequest<LegJson>['legs'][number],
+): IteratorObject<RpcLimitOrder> {
+  if (!leg) return;
+
+  const sellAsset = getInternalAsset(leg.side === 'BUY' ? leg.base_asset : leg.quote_asset);
+
+  for (const { orders, balances } of quotes) {
+    const balance = balances?.[sellAsset];
+
+    for (const [tick, amount] of orders) {
+      if (balance === undefined || isBalanceWithinTolerance(balance, amount)) {
+        yield {
+          LimitOrder: {
+            side: leg.side === 'BUY' ? 'sell' : 'buy',
+            base_asset: leg.base_asset,
+            quote_asset: leg.quote_asset,
+            tick,
+            sell_amount: hexEncodeNumber(amount),
+          },
+        };
+      }
+    }
+  }
+}
+
 type ClientVersion = '2';
+export type AccountId = string & { __brand: 'AccountId' };
```

Comment:
this is to prevent passing random strings right? might be abit confusing if you dont have the context

Reply:
yeah it's a pretty common pattern that we don't use anywhere yet but yeah it's to help keep track of this specific type of string throughout the logic

Reply:
added a comment